src/app/screens/LNURLPay/index.tsx

```diff
@@ -318,6 +319,7 @@ function LNURLPay() {
               <Button
                 onClick={() => {
                   if (successAction.url) utils.openUrl(successAction.url);
+                  msg.reply(paymentResponse);
```

Comment:
Why reply when the user presses a button? :thinking: I am afraid this will lead to inconsistent behavior (whether user chooses to close the popup / clicks "close" button shouldn't make a difference)

@im-adithya Would it be an option to save info (such as the successAction) along with the transaction?

Reply:
if successAction is url. and when user clicks open url button it should redirect user to url as well as close the prompt. so i added it there

Reply:
what happens if the user just closes the popup?

Reply:
> Would it be an option to save info (such as the successAction) along with the transaction?

Ofc, it should be possible because we get the successAction along with the invoice itself 👌 Will add this to the ongoing metadata PRs

Reply:
what happens if the user just closes the popup?

we still resolve the promise with closePromise. that is why its fine

Reply:
https://github.com/getAlby/lightning-browser-extension/blob/f985da8ce4630852ae5f18cf7b2a88c97fbc657b/src/common/lib/utils.ts#L120